chatline.py

```diff
@@ -53,7 +53,7 @@ def is_starting_line(self, line=""):
             (\d{2})     #2 digit of minute
             (\.|:)?     #Zero or one of dot '.' or colon ':'
             (\d{2})?    #Zero or one of 2 digits of second
-            (\s[AP]M)?))  #Zero or one of ('space', 'A' or 'P', and 'M'
+            (\s[apAP][mM])?))  #Zero or one of ('space', 'A' or 'P', and 'M'
```

Comment:
Nice catch @meesha7 !
Keep in mind that the am/pm format may written with or without space, even sometimes written with a dot.
Your code explicitly write a space. Consider something like this `([\s]?[AaPp]?\.?[Mm]?\.?)?)`.

So appreciate it if you can test it :)

Reply:
Thanks! The optional space before am/pm was easy to do and test. However, to make it parse also a.m./p.m. more changes would be needed:

```
diff --git a/chatline.py b/chatline.py
index de47adf..81991de 100644
--- a/chatline.py
+++ b/chatline.py
@@ -53,7 +53,7 @@ class Chatline:
             (\d{2})     #2 digit of minute
             (\.|:)?     #Zero or one of dot '.' or colon ':'
             (\d{2})?    #Zero or one of 2 digits of second
-            (\s?[apAP][mM])?))  #Zero or one of ('space', 'A' or 'P', and 'M'
+            (\s?[apAP]\.?[mM]\.?)?))  #Zero or one of ('space', 'A' or 'P', and 'M'
             (\]?\s-?\s?\s?)#Zero or one close square bracket ']', Zero or one (space and '-'), zero or one space
             (.+)        #One or more character of chat member phone number or contact name
         """
@@ -229,7 +229,7 @@ class Chatline:
             self.is_startingline = True
 
             # Extract timestamp
-            dt = self.extract_timestamp(starting_line.group(2).replace(".", ":"))
+            dt = self.extract_timestamp(starting_line.group(2))
             # Set timestamp
             if dt:
                 self.timestamp = dt
```

Do you think this is ok?

Reply:
Looks perfect 👍

Reply:
Ok, added :) I also added unit tests for the datetime parser. While doing that I noticed that sometimes the date parser interprets the date as MM/DD/YYYY (the American "special" case) instead of DD/MM/YYYY. I'm gonna look into that and make another PR if I find anything :) Think you can merge this now?